packages/nimble-components/src/patterns/error/specs/checkbox-radio-error.md

```diff
@@ -0,0 +1,103 @@
+-   [Nimble issue 2019: Error state for nimble-radio-group](https://github.com/ni/nimble/issues/2019)
+
+Related issues that can and/or should be resolved as part of this effort:
+
+-   [Nimble issue 2091: nimble-checkbox has incorrect height](https://github.com/ni/nimble/issues/2091)
+-   [Nimble issue 2090: Error icon is incorrectly spaced in nimble-text-field](https://github.com/ni/nimble/issues/2090)
+
+## Design
+
+Both the `nimble-checkbox` and the `nimble-radio-group` will be updated to be able to show an error associated with them.
+
+At a high-level, the design of these components will be:
+
+-   Red error text below the component
+    -   The error text does not wrap
+    -   It has an ellipsis if the text is longer than the control
+    -   The `title` of the text is the full error string
+-   Red error icon (`nimble-icon-exclamation-mark`) on the right side of the label with a 4px margin between the icon and the right edge of the control
+-   Control indicator (checkbox square or radio circle) and error icon will both be center-aligned with the first line of label text, both when the label text wraps and when it fits on a single line.
+-   Control indicator and label will both be interactable -- hovering over them will put the component in the "mouseover" state and clicking them will activate the control
+-   The error icon along with the space between the label and the error icon will not be interactable
+
+Below are some examples of how the components will look and behave with different configurations. The blue outline shows the bounds of the control, and the blue background indicates the interactable regions of the control.
+
+![](spec-images/checkbox-no_error,no_wrap.PNG)
+
+![](spec-images/checkbox-no_error,wrap.PNG)
+
+![](spec-images/checkbox-error,no_wrap.PNG)
+
+![](spec-images/checkbox-error,wrap.PNG)
+
+![](spec-images/checkbox-long_error.PNG)
+
+![](spec-images/radio-horizontal,no_error.PNG)
+
+![](spec-images/radio-horizontal,error.PNG)
+
+![](spec-images/radio-vertical,error,no_wrap.PNG)
+
+![](spec-images/radio-vertical,error,wrap.PNG)
+
+![](spec-images/radio-vertical,long_error.PNG)
+
+### API
+
+#### Checkbox
+
+The `nimble-checkbox` will be updated to implement the `ErrorPattern` interface. This means it will have two new attributes:
+
+-   `error-text` - string
+-   `error-visible` - boolean
+
+The `error-visible` boolean will control whether or not a red exclamation icon will be visible next to the checkbox label and whether or not the `error-text` is displayed below the checkbox label.
+
+#### Radio Group
+
+The `nimble-radio-group` will be updated to implement the `ErrorPattern` interface. This means it will have two new attributes:
+
+-   `error-text` - string
+-   `error-visible` - boolean
+
+The `error-visible` boolean will control whether or not a red exclamation icon will be visible next to each radio label and whether or not the `error-text` is displayed below all the radio controls.
```

Comment:
Did we have discussions about whether the exclamation should appear on all the radio labels? My initial reaction from the screenshot with 3 radio buttons is that it looks like 3 separate errors even though there's just one error for the whole control.

It's possible this is the best we can do:
1. If we put the exclamation next to the first or last option I'd probably complain that it looks like only that option has an error. 
2. I don't think there's enough space to put it next to the label.
3. We could decide not to show the exclamation on the radio group and just rely on the error text, but that's inconsistent with other controls.

Just want to raise it in case there's rationale for this approach that we should capture and to see if anyone else has opinions.

Reply:
I believe these alternatives were considered as part of creating the visual design. Perhaps @fredvisser and @NIbokeefe have some context they can provide here.

Reply:
> looks like 3 separate errors even though there's just one error for the whole control.

+1 for that concern. I also expect the interaction would feel pretty jarring for all of the indicators to appear / disappear at the same time.

Reply:
@fredvisser or @NIbokeefe, can you weigh in here on more of the rationale of the design and whether or not we want to revisit the current visual design?